renderdoc/os/posix/android/android_stringio.cpp

```diff
@@ -63,15 +64,33 @@ namespace FileIO
 {
 const char *GetTempRootPath()
 {
-  return "/sdcard";
+  static string ret;
+  GetExecutableFilename(ret);
+
+  // This folder is writable even if the APK does not have manifest write permissions.
+  ret = "/data/data/" + ret + "/files";
+  return ret.c_str();
 }
 
+// For RenderDocCmd.apk, this returns "org.renderdoc.renderdoccmd"
+// For other APKs, we use it to get the writable temp directory.
 void GetExecutableFilename(string &selfName)
 {
-  char path[512] = {0};
-  readlink("/proc/self/exe", path, 511);
+  char buf[4096];
+  snprintf(buf, sizeof(buf), "/proc/%u/cmdline", getpid());
```

Comment:
I was thinking that if there could be command line parameters, we'd need to parse out just the executable name here to be correct (otherwise we'd be returning the whole command line as the executable name). If the common case is not to have a command line though, then this is fine.